Source/WebGPU/WGSL/AST/ASTIdentifierExpression.h

```diff
@@ -46,6 +47,12 @@ class IdentifierExpression final : public Expression {
 
 private:
     String m_identifier;
+
+protected:
+    // FIXME: This is necessary becasue we replace an IdentifierExpression with
+    // a StructureAccess for globals, so the sizes must be compatible. We need to
+    // introduce an IdentityExpression and remove this hack
+    uint8_t m_padding[sizeof(StructureAccess) - sizeof(Expression) - sizeof(String)];
```

Comment:
I think if we use an identity node as I describe in the FIXME it's pretty safe to use placement new? One thing I was considering is moving the AST representation to be a little more structured as an IR, so that modifications can become easier. That would make a lot of things easier.